daml/Marketplace/Custody/Model.daml

```diff
@@ -113,9 +114,13 @@ template LifecycleRequest
               election : Claim Observation Date Id -> Claim Observation Date Id -> Claim Observation Date Id
               election _ _ = deserialize choice -- will throw if the claims don't match
           Lifecycle.Result{remaining, pending} <- Lifecycle.lifecycle spot election (deserialize description.claims) t
-          (_, postAssetVersion) <- exercise versionsCid . AssetDescription.LookupOrInsert $ serialize remaining
-          payoutCids <- forA pending \(qty, asset) -> do
-            payoutDepositCid :: _ <- exercise safekeepingDepositCid $ AssetDeposit_Split [qty * assetDeposit.asset.quantity]
+          (_, postAssetVersion, _) <- exercise versionsCid AssetDescription.LookupOrInsert with newClaims = serialize remaining, assetDescription = description
+          let payoutSplitQuantities = (\(qty, _) -> qty * assetDeposit.asset.quantity) <$> pending
+          -- Next step could be:
+          -- create SettlementInstruction with buyer = customer; seller = issuer; ..
+          splitResultDeposits <- exercise safekeepingDepositCid AssetDeposit_Split with quantities = payoutSplitQuantities
+          let payoutCids = init splitResultDeposits
+          payoutCids <- forA payoutCids \payoutDepositCid -> do
```

Comment:
Just my opinion, but I think the de-structuring is more legible than using `init` here. (Plus you save an import :wink: )

Reply:
May I ask how exactly you mean de-structuring? I am probably missing something but in this case 'we need everything but the last element' from the list.

Reply:
So if you look at the diff, this is replacing
```
payoutDepositCid :: _ <- exercise safekeepingDepositCid $ AssetDeposit_Split [qty * assetDeposit.asset.quantity]
```
The right-hand-side of the arrow is taking the head of the tail for you. In this case `_` is the result of `init`.

Reply:
Oh sorry, my bad here, I thought you were taking the tail. Pls ignore this comment.